### PR TITLE
Example of building a TopoJSON object and writing it to a string

### PR DESCRIPTION
**Pull request by Priceless-P**

### CODE REVIEW

**Comment by Priceless-P** [comment]
@patricoferris please the `Object` okay? Should I proceed with writing it to a string?

**Comment by Priceless-P** [comment]
@patricoferris please the `Object` okay? Should I proceed with writing it to a string?

File: src/topojson/topojson_intf.ml [hunk 1]
```diff
@@ -264,7 +264,7 @@ module type S = sig
 
   val topojson : t -> topojson
   val bbox : t -> float array option
-
+  val vals_of_t : topojson -> topojson -> float array option -> t
```
Comment (patricoferris) [inline src/topojson/topojson_intf.ml]: Ah yes we need to expose constructors for everything in order to build a new value. For the main topojson object we will need:

```ocaml
val v : ?bbox:float array -> topojson -> t
``` 

And then we will need constructors for building `topojson` values too.

File: README.md [hunk 2]
```diff
@@ -90,10 +90,32 @@ A small example how this library is efficient in reading a `json` file _(more pa
...
-# Topojson.of_json ( Ezjsonm.value_from_string topojson_string);; 
+# Topojson.of_json ( Ezjsonm.value_from_string topojson_string);;
 - : (Topojson.t, [ `Msg of string ]) result = Ok <abstr>
 ```
 
+### Example of building a TopoJSON object and writing it to a string
+
+```ocaml
+let topojson =
+  let objects = [ { "example": "type" = "GeometryCollection"; geometries = [{
```
Comment (patricoferris) [inline README.md]: We will need to use functions to construct these values. At the moment this is a mix of JSON/OCaml which will not compile, see my other comment.

**Review by patricoferris — COMMENTED** [review]
Thanks! We haven't exposed/implemented enough functions to construct values to do this properly yet.

**Comment by Priceless-P** [comment]
@patricoferris please check again. How do I proceed from here?

**Comment by patricoferris** [comment]
Awesome! Great work, to help get this over the line, I hopefully got it to a working state in this commit https://github.com/geocaml/ocaml-topojson/commit/4765b65f618789681171c83481a068726814ccef -- feel free to incorporate that work if it is helpful and please ask any questions you might have :))

**Comment by Priceless-P** [comment]
> feel free to incorporate that work if it is helpful 

Done, thank you.

So we add the signatures to the markup file too. I want to confirm what this `= <abstr>` means in
```ocaml
val t : Topojson.t = <abstr>
```

**Comment by patricoferris** [comment]
We just need a `dune build @fmt` to get the tests all green :))

```ocaml
val t : Topojson.t = <abstr>
```

This means that the type is abstract and that the top-level (which MDX uses) doesn't have a registered printer function to show the actual value. That's completely fine.

**Comment by Priceless-P** [comment]
>This means that the type is abstract and that the top-level (which MDX uses) doesn't have a registered printer function to show the actual value. That's completely fine.

Okay. Thanks for explaining that. I have formatted the code too.

**Comment by patricoferris** [comment]
Great thanks!

**Pull request merged**